sqlglot/parser.py

```diff
@@ -3532,7 +3532,7 @@ def _parse_case(self) -> t.Optional[exp.Expression]:
 
     def _parse_if(self) -> t.Optional[exp.Expression]:
         if self._match(TokenType.L_PAREN):
-            args = self._parse_csv(self._parse_conjunction)
+            args = self._parse_csv(self._parse_expression)
```

Comment:
are ternary farther up the tree than conjunction? could conjunction call tern?

Reply:
That's how I had originally implemented it, but moved it one level above it because the logic cut across several methods and so was too invasive.

Reply:
but what is correct? what has more precedence

Reply:
Conjunction binds more tightly, i.e. `a and b ? foo : bar` is to be parsed as `(a and b) ? foo : bar`, so the current approach is reasonable with respect to that.

The only current issue is that in order to handle ternaries whenever we'd have to use `_parse_expression` instead of conjunction, unless we mix ternary and conjunction logic into `_parse_conjunction`, depending on whether `?` is matched. I'm not yet sure how straightforward this change will be, but could be an easy fix if you think it's worth it.